File: app/src/main/java/com/wafflestudio/snutt_staging/view/ToggleRadioButton.java
package com.wafflestudio.snutt_staging.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RadioButton;
import android.widget.RadioGroup;

/**
 * Created by makesource on 2017. 8. 27..
 */

public class ToggleRadioButton extends android.support.v7.widget.AppCompatRadioButton {

    public ToggleRadioButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public void toggle() {
        if(isChecked()) {
            if(getParent() instanceof RadioGroup) {
                ((RadioGroup)getParent()).clearCheck();
            }
        } else {
            setChecked(true);
        }
    }
}

File: app/src/main/java/com/wafflestudio/snutt2/view/ToggleRadioButton.kt
package com.wafflestudio.snutt2.view

import android.content.Context
import android.util.AttributeSet
import android.widget.RadioGroup
import androidx.appcompat.widget.AppCompatRadioButton

/**
 * Created by makesource on 2017. 8. 27..
 */
class ToggleRadioButton(context: Context?, attrs: AttributeSet?) : AppCompatRadioButton(context, attrs) {
    override fun toggle() {
        if (isChecked) {
            if (parent is RadioGroup) {
                (parent as RadioGroup).clearCheck()
            }
        } else {
            isChecked = true
        }
    }
}